### PR TITLE
Add documentation for core Docker concepts and guides

### PR DESCRIPTION
**Pull request by ajeetraina**
## Description

This PR introduces a new collection of guides aimed at establishing a strong foundation in core Docker concepts. The initial set of guides focuses on:

- What is a container?
- What is an image?
- What is a registry?
- What is Docker Compose?

### CODE REVIEW

**Comment by mikesir87** [comment]
Just rebased on top of current `main` to resolve conflicts, etc.

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+## Try it now
+
+In this hands-on, you will learn how to build and push a Docker image to the Docker Hub repository.
+
+### Sign up for a free Docker account
+
+1. If you have not done so, [create a Docker account](https://hub.docker.com/signup).
+
+    ![Screenshot of the official Docker Hub page showing the Sign up page](images/dockerhub-signup.webp?border)
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-a-registry.md:37]: Suggest updating the image. For me, the image doesn't match where the link took me.

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    You can use your Google or GitHub account to authenticate.
+
+### Create your first repository
+
+1. Sign in to [Docker Hub](https://hub.docker.com)
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-a-registry.md:43]:
```suggestion
1. Sign in to [Docker Hub](https://hub.docker.com).
```

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+2. Select the **Create repository** button in the top-right corner
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-a-registry.md:44]:
```suggestion
2. Select the **Create repository** button in the top-right corner.
```

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    ![Screenshot of the Docker Hub page that shows how to create a public repository](images/create-hub-repository.webp?border)
+
+4. Set the visibility to **Public**. 
+5. Select the **Create** button to create the repository.
+
+That's it. You've successfully created your first repository. 🎉
+
+This repository is empty right now. You'll now fix this by pushing an image to it.
+
+### Login with Docker Desktop
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-a-registry.md:56]:
```suggestion
### Sign in with Docker Desktop
```

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+1. [Download and install](https://www.docker.com/products/docker-desktop/) Docker Desktop, if not already installed.
+2. In the Docker Desktop GUI, select the **Log in** button in the top-right corner
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-a-registry.md:59]:
```suggestion
2. In the Docker Desktop GUI, select the **Sign in** button in the top-right corner
```

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+### Cloning a sample Node.js code
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-a-registry.md:61]: Nit: Suggest using the same conjugation as the previous task-based sections. 

```suggestion
### Clone sample Node.js code
```
Or `Clone a sample Node.js project`

File: content/guides/docker-concepts/the-basics/what-is-docker-compose.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+description: What is Docker Compose?
+---
+
+{{< youtube-embed xhcUIK4fGtY >}}
+
+## Explanation
+
+If you've been following the guides so far, you've been working with single container applications. But, now you're wanting to do something more complicated - run databases, message queues, caches, or a variety of other services. Do you install everything in a single container? Run multiple containers? If you run multiple, how do you connect them all together?
+
+One container best practice is that each container should do one thing and do it well. While there are exceptions to this rule, avoid the tendency to have one container do multiple things.
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-docker-compose.md:13]: I tripped up reading this the first time. Maybe reword to:

```suggestion
One best practice for containers is that each container should do one thing and do it well. While there are exceptions to this rule, avoid the tendency to have one container do multiple things.
```

File: content/guides/docker-concepts/the-basics/what-is-docker-compose.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    >
+    > Note that if you remove the containers for a Compose app in the GUI, it's removing only the containers. You'll have to manually remove the network and volumes if you want to do so.
+    { .important }
+
+In this walkthrough, you learned how to use Docker Compose to start and stop a multi-container application.
+
+
+## Additional resources
+
+This concept page purposely didn't do a deep-dive on Compose and how to write a Compose file, but focused on introducing you to what Compose is.
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-docker-compose.md:132]: Maybe reword to have a CTA.

```suggestion
This page was a brief introduction Compose. In the following resources, you can dive deeper into Compose and how to write a Compose files.
```

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+Now that you know what a container image is and how it works, you might wonder - where do you store these images? 
+
+Well, you can store your container images on your computer system, but what if you want to share them with your friends or use them on another machine? That's where the image registry comes in.
+
+An image registry is a centralized location for storing and sharing your container images. It can be either public or private. [Docker Hub](https://hub.docker.com) is a public registry that anyone can use and is the default registry. 
+
+While Docker Hub is a popular option, there are many other available container registries available today, including [Amazon Elastic Container Registry(ECR)](https://aws.amazon.com/ecr/), [Azure Container Registry (ACR)](https://azure.microsoft.com/en-in/products/container-registry), and [Google Container Registry (GCR)](https://cloud.google.com/artifact-registry). You can even run your private registry on your local system or inside your organization. For example, Harbor, JFrog Artifactory, GitLab Container registry etc.
+
+### Registry vs repository
```
Comment (craig-osterhout) [inline content/guides/docker-concepts/the-basics/what-is-a-registry.md:19]:
```suggestion
### Registry vs. repository
```
Or `versus`.

File: content/guides/docker-concepts/the-basics/what-is-a-registry.md (hunk 1 above)
Comment (ajeetraina) [reply content/guides/docker-concepts/the-basics/what-is-a-registry.md:37]: I adjusted the wording to align with the image. Hope it looks good.

File: content/guides/docker-concepts/the-basics/what-is-an-image.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    >
+    > If you add the `--no-trunc` flag to the command, you will see the full command. Note that, since the output is in a table-like format, longer commands will cause the output to be very difficult to navigate.
+    { .tip }
+
+In this walkthrough, you searched and pulled a Docker image. In addition to pulling a Docker image, you also learned about the layers of a Docker image.
+
+{{< /tab >}}
+{{< /tabs >}}
+
+In this walkthrough, you searched and pulled a Docker image. In addition to pulling a Docker image, you also learned about the layers of a Docker Image.
```
[the remark this answers is not in the document]
Comment (ajeetraina) [reply content/guides/docker-concepts/the-basics/what-is-an-image.md:189]: Thanks for pointing this out. Fixed it.

File: content/guides/docker-concepts/the-basics/what-is-docker-compose.md (hunk 8 above)
Comment (ajeetraina) [reply content/guides/docker-concepts/the-basics/what-is-docker-compose.md]: Adjusted the wording.

File: content/guides/docker-concepts/the-basics/what-is-a-container.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+## Additional resources
+
+The following links provide additional guidance into containers:
+
+- [Running a container](/engine/reference/run/)
+- [What is a container?](/guides/walkthroughs/what-is-a-container/)
+- [Overview of container](https://www.docker.com/resources/what-container/)
+- [Why Docker?](https://www.docker.com/why-docker/)
+
+Now that you have learned the basics of a Docker container, it's time to learn about Docker images.
```
[the remark this answers is not in the document]
Comment (ajeetraina) [reply content/guides/docker-concepts/the-basics/what-is-a-container.md:189]: Thanks for the suggestion. Added "Next Steps".

**Comment by ajeetraina** [comment]
@craig-osterhout I appreciate the feedback in your review. I've made all the requested changes. Please let me know if you have any further comments.

**Comment by craig-osterhout** [comment]
LGTM 🚀
Regarding the comment about the prompt ($) in front of the command, my meaning was to add it to all the commands in codeblocks for consistency. Not a blocker though, we can clean it up in a follow-up.

**Review by craig-osterhout — APPROVED** [review]
